File: Data/C++/linear search.cpp
#include <iostream>
#include <vector>
using namespace std;

// Linear search checks every element
int linearSearch(const vector<int>& nums, int target) {
    for (int i = 0; i < nums.size(); i++) {
        if (nums[i] == target)
            return i;  // Found at index i
    }
    return -1;  // Not found
}

// ✅ Use-case
int main() {
    vector<int> nums = {4, 2, 9, 1, 6};
    int result = linearSearch(nums, 1);
    cout << "Index of 1: " << result << endl;  // Output: 3
}

File: Data/Python/linear search.py
def linear_search(arr, target):
    """
    Linear Search Algorithm:
    - Iterates through every element to find the target.
    - Works on unsorted arrays.
    - Time Complexity: O(n)
    """
    # Traverse each index in the array
    for i in range(len(arr)):
        # Check if current element is the target
        if arr[i] == target:
            return i  # Return index if found
    return -1  # Not found

# ✅ Example
nums = [10, 2, 8, 4]
print("Linear Search Result:", linear_search(nums, 8))  # Output: 2
